lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -102,17 +110,29 @@ defmodule KinesisClient.Stream.Shard.Producer do
     get_records(%{state | poll_timer: nil})
   end
 
-  def handle_info(:shard_closed, %{coordinator_name: coordinator, shard_id: shard_id} = state) do
-    # just in case something goes awry, try and close the Shard in the future
+  def handle_info(
+        :shard_closed,
+        %{coordinator_name: coordinator, shard_id: shard_id, child_shards: child_shards} = state
+      ) do
     Logger.info(
       "Shard is closed, notifying Coordinator: [app_name: #{state.app_name}, " <>
         "shard_id: #{state.shard_id}]"
     )
 
-    timer = Process.send_after(self(), :shard_closed, state.shutdown_delay)
-    AppState.close_shard(state.app_name, state.shard_id, state.app_state_opts)
-    :ok = Coordinator.close_shard(coordinator, shard_id)
-    {:noreply, %{state | shard_closed_timer: timer}}
+    :ok =
+      AppState.close_shard(
+        state.app_name,
+        shard_id,
+        state.lease_owner,
+        state.app_state_opts
+      )
+
+    :ok = Coordinator.append_shards(coordinator, child_shards)
+
+    Shard.name(state.app_name, state.stream_name, shard_id)
```

Comment:
This is the part that we need to figure out when/how we handle shutdown, allowing for the broadway pipeline to drain.

Reply:
Looks like Broadway's shutdown sequence handles this for us. All good.